numba/cuda/compiler.py

```diff
@@ -309,15 +310,15 @@ def inspect_ptx(self, args, nvvm_options={}):
         -------
         ptx : bytes
         """
-        llvmir = self.inspect_llvm(args)
-        # Make PTX
-        cuctx = get_context()
-        device = cuctx.device
-        cc = device.compute_capability
-        arch = nvvm.get_arch_option(*cc)
-        opt = 3 if self.opt else 0
-        ptx = nvvm.llvm_to_ptx(llvmir, opt=opt, arch=arch, **nvvm_options)
-        return ptx
+        msg = ('inspect_ptx for device functions is deprecated. Use '
+               'compile_ptx instead.')
+        warn(msg, category=NumbaDeprecationWarning)
+
+        if nvvm_options:
+            msg = ('nvvm_options are ignored. Use compile_ptx if you want to '
+                   'set NVVM options.')
+            warn(msg, category=NumbaDeprecationWarning)
+        return self.compile(args).library.get_asm_str().encode()
```

Comment:
Add a note in the deprecation notices?

Reply:
Done.

Reply:
Thanks. Doc build https://numba--6953.org.readthedocs.build/en/6953/reference/deprecation.html#deprecation-of-the-inspect-ptx-method looks good.